mne/preprocessing/tests/test_ica.py

```diff
@@ -705,7 +705,8 @@ def test_ica_additional(method, tmpdir, short_raw_epochs):
     assert_equal(ica_sorted.labels_, dict(blink=[3], think=[2]))
 
     # epochs extraction from raw fit
-    pytest.raises(RuntimeError, ica.get_sources, epochs)
+    with pytest.warns(RuntimeWarning, match='could not be picked'):
+        pytest.raises(RuntimeError, ica.get_sources, epochs)
```

Comment:
`pytest.raises` is a context manager, so better to do:
```
with pytest.warns(...):
    with pytest.raises(RuntimeError, match='some matching regex'):
         ...
```
if you want to avoid so much nesting you could do:
```
warn_ctx = pytest.warns(RuntimeWarning, match='...')
err_ctx = pytest.raises(RuntimeError, match='...'):
with warn_ctx, err_ctx:
    ...
```